cpp/src/arrow/dataset/file_parquet_test.cc

```diff
@@ -88,5 +94,105 @@ TEST_F(TestParquetFileFormat, ScanRecordBatchReader) {
   ASSERT_EQ(row_count, kNumRows);
 }
 
+class TestFileSystemBasedDataSource : public FileSourceFixtureMixin {
```

Comment:
Is it me or is this test class largely duplicated in `file_test.cc`? You might want to factor out the common parts, perhaps in a base class.